### PR TITLE
fix(Core/Deps): Update recastnavigation to last version 

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
thanks @jackpoz for pointing this out

-----------------

## How to test:

- Checkout this branch and recompile, including recompiling the client data files extractor tools (maps/dbc/vmaps/mmaps )
- Extract the data files (use a clean client) and run a server with the newly extracted data (enable vmaps/mmaps)
- Log in the game and have a look around to make sure the maps work fine. You could check, for example, stuff like elevators, pathing, casting spells from behind walls, etc...

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
need to solve:

```
/home/travis/build/azerothcore/azerothcore-wotlk/src/tools/mmaps_generator/MapBuilder.cpp:462:24: fatal error: use of undeclared identifier 'DT_POLY_BITS'
        int polyBits = DT_POLY_BITS;
                       ^
1 error generated.
src/tools/mmaps_generator/CMakeFiles/mmaps_generator.dir/build.make:75: recipe for target 'src/tools/mmaps_generator/CMakeFiles/mmaps_generator.dir/MapBuilder.cpp.o' failed
make[2]: *** [src/tools/mmaps_generator/CMakeFiles/mmaps_generator.dir/MapBuilder.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

**Comment by Viste** [comment]
first we can remove #include <ace/OS_NS_unistd.h> from defines in MapBuilder.cpp(dont need)

and change includes to
```
#include "MapBuilder.h"
#include "MapTree.h"
#include "ModelInstance.h"
#include "PathCommon.h"

#include <DetourCommon.h>
#include <DetourNavMesh.h>
#include <DetourNavMeshBuilder.h>

#include "DisableMgr.h"```

**Comment by jackpoz** [comment]
You should apply https://github.com/TrinityCore/TrinityCore/blob/3.3.5/dep/recastnavigation/recastnavigation.diff on top of recastsources and maybe include it in your repo too.

**Comment by FrancescoBorzi** [comment]
Thanks @Viste and @jackpoz I'll have a look at your suggestions

**Comment by FrancescoBorzi** [comment]
@jackpoz I've updated it with your patch, however the build is still failing with:

```
src/server/scripts/CMakeFiles/scripts.dir/build.make:7160: recipe for target 'src/server/scripts/CMakeFiles/scripts.dir/Pet/pet_mage.cpp.o' failed
make[2]: *** [src/server/scripts/CMakeFiles/scripts.dir/Pet/pet_mage.cpp.o] Terminated
src/server/scripts/CMakeFiles/scripts.dir/build.make:7173: recipe for target 'src/server/scripts/CMakeFiles/scripts.dir/Pet/pet_priest.cpp.o' failed
make[2]: *** [src/server/scripts/CMakeFiles/scripts.dir/Pet/pet_priest.cpp.o] Terminated
src/server/scripts/CMakeFiles/scripts.dir/build.make:7186: recipe for target 'src/server/scripts/CMakeFiles/scripts.dir/Pet/pet_shaman.cpp.o' failed
```

https://travis-ci.org/azerothcore/azerothcore-wotlk/builds/569525253

Also note that the build with clang 7 is passing (build id 2) but the clang 3.8 (build id 1) is failing with that error

**Comment by Stoabrogga** [comment]
@FrancescoBorzi 
The Travis job just took to long, that's why it was terminated. I started it again.

**Comment by FrancescoBorzi** [comment]
oh nice, thank you @Stoabrogga !

**Comment by Stoabrogga** [comment]
Tested this PR together with #2190. So far successfully:
- extractor tools worked
- tested #923
- tested the elevator in the Undercity
- tested a few creatures with waypoint paths (kodos in the barrens, Dalaran visitors, racing at the Mirage Raceway)
- played through Pit of Saron
- tried to cast from behind a wall

But I think you should increased the MMAP and VMAP version numbers here:
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/common/Collision/VMapDefinitions.h#L15-L16
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Miscellaneous/SharedDefines.h#L3523
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/tools/mesh_extractor/Utils.cpp#L28
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/tools/mesh_extractor/Utils.h#L343
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/tools/mmaps_generator/MapBuilder.cpp#L26
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/tools/vmap4_extractor/vmapexport.cpp#L67

**Comment by Stoabrogga** [comment]
I pushed a new commit and increased the MMAP and VMAP version numbers.

**Comment by Winfidonarleyan** [comment]
Maybe rename the title to `feat(Core/Deps): Update recastnavigation to last version` ?

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan done, but this is a fix and not a new feature.

I will merge this and update the data files tonight.

**Comment by Winfidonarleyan** [comment]
Okay

File: deps/recastnavigation/Detour/Include/DetourMath.h [hunk 1]
```diff
@@ -16,5 +19,6 @@ inline float dtMathCeilf(float x) { return ceilf(x); }
 inline float dtMathCosf(float x) { return cosf(x); }
 inline float dtMathSinf(float x) { return sinf(x); }
 inline float dtMathAtan2f(float y, float x) { return atan2f(y, x); }
+inline bool dtMathIsfinite(float x) { return std::isfinite(x); }
```
Comment (pklloveyou) [inline deps/recastnavigation/Detour/Include/DetourMath.h]: //inline bool dtMathIsfinite(float x) { return std::isfinite(x); }
inline bool dtMathIsfinite(float x) { return isfinite(x); }

Comment (FrancescoBorzi) [reply deps/recastnavigation/Detour/Include/DetourMath.h]: @pklloveyou what error? I don't understand

Comment (pklloveyou) [reply deps/recastnavigation/Detour/Include/DetourMath.h]: @FrancescoBorzi Cannot compile

Comment (pklloveyou) [reply deps/recastnavigation/Detour/Include/DetourMath.h]: Mmpas are being extracted.Let me try.

Comment (FrancescoBorzi) [reply deps/recastnavigation/Detour/Include/DetourMath.h]: @pklloveyou what compiler do you use and which version?

Comment (pklloveyou) [reply deps/recastnavigation/Detour/Include/DetourMath.h]: Thank you.

Comment (pklloveyou) [reply deps/recastnavigation/Detour/Include/DetourMath.h]: @FrancescoBorzi VS2019

Comment (FrancescoBorzi) [reply deps/recastnavigation/Detour/Include/DetourMath.h]: @Mik1893 can you reproduce this? Because it works fine for me on Ubuntu

Comment (pklloveyou) [reply deps/recastnavigation/Detour/Include/DetourMath.h]: Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.14393.
The C compiler identification is MSVC 19.22.27905.0

Comment (mik1893) [reply deps/recastnavigation/Detour/Include/DetourMath.h]: inline bool dtMathIsfinite(float x) { return isfinite(x); }

drop std::

**Comment by sudlud** [comment]
recastnavigation/recastnavigation@c40188c796f089f89a42e0b939d934178dbcfc5c

**Pull request merged**